sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -335,7 +335,7 @@ def _incr_mean_variance_axis0(np.ndarray[floating, ndim=1] X_data,
     # Next passes
     for i in range(n_features):
         updated_n[i] = last_n[i] + new_n[i]
-        last_over_new_n[i] = last_n[i] / new_n[i]
+        last_over_new_n[i] = dtype(last_n[i]) / dtype(new_n[i])
```

Comment:
Would this not also affect other cases where integer division was not appropriate?

Can we make sure we test the case where new_n[I] is 0 if we do not already?

Reply:
This might happen if we have only NaN values in a column when doing a `partial_fit`. It is fair to investigate this case. Thanks.

Reply:
We should also test that the computed variance:
- is the same for sparse and dense
- is correct, by comparing with `np.var(np.vstack((X1,X2)))` on the dense arrays (assuming numpy is correct),
- with partial fit
- on data with n_samples that don't divide each other.